packages/swing-store/src/bundleStore.js

```diff
@@ -62,20 +66,36 @@ export function makeBundleStore(db, ensureTxn, noteExport = () => {}) {
     return `b${Nat(version)}-${hash}`;
   }
 
-  const sqlAddBundle = db.prepare(`
-    INSERT OR REPLACE INTO bundles
-      (bundleID, bundle)
-    VALUES (?, ?)
+  // the PRIMARY KEY constraint requires the bundleID not already
+  // exist
+  const sqlAddBundleRecord = db.prepare(`
+INSERT INTO bundles (bundleID, bundle) VALUES (?, NULL)
   `);
 
-  /**
-   * Store a bundle.  Here the bundle itself is presumed valid.
-   *
-   * @param {string} bundleID
-   * @param {Bundle} bundle
-   */
-  function addBundle(bundleID, bundle) {
+  // this sees both populated and pruned (not-yet-populated) records
+  const sqlHasBundleRecord = db.prepare(`
+    SELECT count(*)
+    FROM bundles
+    WHERE bundleID = ?
+  `);
+  sqlHasBundleRecord.pluck();
+
+  const sqlPopulateBundleRecord = db.prepare(`
+    UPDATE bundles SET bundle = $serialized WHERE bundleID = $bundleID
+  `);
+
+  function addBundleRecord(bundleID) {
+    ensureTxn();
+    sqlAddBundleRecord.run(bundleID);
+  }
+
+  function populateBundle(bundleID, serialized) {
     ensureTxn();
+    sqlHasBundleRecord.run(bundleID) || Fail`missing ${bundleID}`;
+    sqlPopulateBundleRecord.run({ bundleID, serialized });
```

Comment:
Would it make sense to do this in one shot?
```suggestion
    const { changes } = sqlPopulateBundleRecord.run({ bundleID, serialized });
    changes > 0 || Fail`missing ${bundleID}`;
```

Reply:
Nope, the first statement is asserting that we already have the metadata present. If we combined them, then even with the `changes > 0` check, we'd be allowing the repopulation to invent new metadata that wasn't previously established. That would allow the set of artifacts to create a new truth, when repopulation is supposed to be limited to previously-accepted truths.

Changing `addBundle` to do everything in a single step would be ok, but the call from `importBundle` would not (even during an import, we treat the export-data as truth, and the artifact/artifact-names as untrusted). And when we later implement `repopulateBundle`, the issue will really stand out.

Also, I don't really want to rely upon the SQLite binding's behavior so much.. a casual reader can guess that `run()` takes a statement, but it's not very obvious what a `{ changes }` return value might mean.